BIDScramble/bidscramble/scramble_fif.py

```diff
@@ -50,10 +34,18 @@ def scramble_fif(inputdir: str, outputdir: str, select: str, bidsvalidate: bool,
 
         # Apply the scrambling method
         if method == 'permute':
-            obj.apply_function(do_permute)
+            axis = dict([(d, i) for i, d in enumerate(['channel', 'time'])])
+
+            if not type(dims) is list:
+                dims = [dims]
+
+            # scramble the samples across the requested dimension(s)
+            rng = np.random.default_rng()
+            for dim in range(len(dims)):
+                rng.shuffle(obj._data, axis=axis[dims[dim]])
 
         elif method in ('null', None):
-            obj.apply_function(do_null)
+            obj._data *= 0
```

Comment:
It's probably ok but in general, variables that begin with an underscore are private / not to be used outside the function in which they are defined.

Reply:
Yes, I know, but this is a trick that Britta suggested to me, and actually uses herself all the time. With my limited Python knowledge, it's either 'this' (i.e. the ._data syntax), or back to the apply_function